### PR TITLE
Add pre commit hook

### PR DESCRIPTION
**Pull request by xmnlab**
This PR:

- adds pre-commit hook for flake8, mypy, isort and black
- fixes code style using black
- adds black and isort configuration
- adds `make develop` command that prepare and install anything necessary for the development (current just install pre-commit hooks).

### CODE REVIEW

**Comment by xmnlab** [comment]
@datapythonista when you have time, this PR is ready for review. thanks!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -83,3 +83,7 @@ jobs:
     - name: isort
       run: isort --check-only .
       if: always()
+
+    - name: pre-commit hook check
```
Comment (xmnlab) [inline .github/workflows/main.yml:87]: maybe, using the pre commit for lint we can remove the step for flake8, mypy and isort.

I had some problems in the past with pydocstyle as pre-commit hook, but I can create a follow up PR for that to check if it is working now.

**Comment by datapythonista** [comment]
Thanks @xmnlab, many things mixed here. If you want to introduce all those things, please:
- Create one issue for each (or a PR if it's trivial to wirte), and let's discuss first
- Open a separate PR for each issue after we agreed to move forward

But personally, not a big fan of black or pre-commit, and iirc they are not being in other OmniSci projects (I think I checked pymapd, not sure if others). And also -1 in adding a Makefile to a Python project. But feel free to open an issue explaining what value do you think all those things will add to this project, and let's discuss. Closing this if you don't mind, and let's approach these things separately.

**Comment by xmnlab** [comment]
yep black and pre-commit are already in pymapd

https://github.com/omnisci/pymapd/blob/master/pyproject.toml
https://github.com/omnisci/pymapd/blob/master/.pre-commit-config.yaml


ok I will split the PR.  thanks

**Pull request closed without merging**